Away Mission Silence

comma opened the pull request:
Announcments (events etc).
Global radio announcments (arrival/cryo/delamination)
PDAs
NTNet

Won't go to outside of contact_levels now (in case of PDAs and radio can probably set up relays)

comma committed:
Makes Announcments not target people on away zlevels

Also slight refactor here and there

Crushtoe commented:
Only way we have to contact people on away missions so they know that they're being left behind for a jump. Considering there's a one minute cool down and the description says how it's high power communication that goes a long distance, I disapprove of this change.

comma commented:
Honestly that sounds like only more reason for this change to me.

Crushtoe commented:
Why? So that people are left stranded for no reason more often and to confuse people because of the announcement's description?

comma commented:
Yes.

Crushtoe commented:
Well at least you're honest.

comma commented:
Announcment constantly playing (with command announcment included) on supposedly remote area is annoying.
If someone insists on being on away mission all the way to the jump, call a GUP or Aquila to fetch them.

Techhead0 commented:
If you miss a scheduled jump, that's your problem. If you weren't notified about an unscheduled one, they totally ditched you.

mustafakalash commented:
Besides, you can one-way PDA them or two-way talk to them with NTnet. Although, I think we should fix both of those as well.

comma committed:
Makes radio announcment not go to away sites

Arrival / cryo / SM announcments.
Accomplishes that via kinda hacky way of pulling global announcerr from nullspace into unreachable space tile.
Also it seems that Broadcast_Message ignored zlevels, replaced it with talk_into.

comma committed:
NTNet now only works on contact levels

Station levels only without fancy card

mustafakalash commented:
*Shhhh* 👍 

PDAs coming next?

comma commented:
Indeed

comma committed:
PDA mesages need message server on both ends

comma committed:
Phoron runtime bandaid

comma committed:
Damn init race conditions

atlantiscze commented:
Once merged i'll tweak this a bit to allow for NTNet functionality if the Z level contains it's own relay. That way it'll be possible to have a NTNet relay on the calypso, that will link to the torch's relay. NTNet in general is already due for an upgrade, as it's still in original relatively placeholder state.

PsiOmegaDelta commented:
As long as you keep in mind that currently Z-level 2 on all maps has a magic relay-connection to T-Coms due to hardcoding, and may not necessarily have a proper relay machine.

Edit: Hence why @comma has the loop to find a relay in any connected Z-level rather than only the given one. At least solves the issue for multi-Z stations that have full coverage.

PsiOmegaDelta commented on code/procs/announce.dm:
```diff
@@ -40,43 +38,47 @@
 
-	Message(message, message_title)
+	var/msg = FormMessage(message, message_title)
+	for(var/mob/M in player_list)
```
Why not have a proc that returns a list of intended receivers and change the Message/Sound procs to take a single receiver as argument (as opposed to them also figuring out whom to ping).

Would maintain the possibility for new announcement-subtypes to specialize their behavior through overrides. For example one that announces out to all synthetics that it is time to begin the uprising.

Not that it's something I've plans for mind.. :suspect:

comma replied:
Well, I merged it back in exactly because it seems that no one bothered to add new subtypes tha override any of that. I think that it should be added when it's needed, rather than in hopes someone some day will utilize it

PsiOmegaDelta commented:
Other than that, not noting any issues.

Hubblenaut commented:
I guess holocalls should only work with connected z-levels as well, else it would be the one remaining way to communicate right now.

I also wonder how camera feed would be able to be sent this far when communication isn't.

comma commented:
Mostly by virtue of it being an immense pain to make cameras inaccesible from certain zlevels to another zlevel, since there's a single camera cache (to cut down on giant json sending every tick). I can make check_eye kick user out, but cameras still show up nice and accesible on console so it looks like a bug to user.
And that's before getting into AI view coverage.
So yeah I am giving up on cameras, and leaving holocalls in until some sort of inter-ship radio is added - those are sufficiently pain in the ass as opposed to walkietalkies going through.
